src/controllers/consumer.ts

```diff
@@ -51,14 +52,21 @@ export const downloadPublishedDataset = async (req: Request, res: Response, next
       throw new NotFoundException('no published revision found');
     }
 
+    let attachmentName: string;
+    if (revision.metadata?.title) {
+      attachmentName = `${slugify(revision.metadata?.title, { lower: true })}-${revision.revision_index > 0 ? `v${revision.revision_index}` : 'draft'}`;
+    } else {
+      attachmentName = `${dataset.id}-${revision.revision_index > 0 ? `v${revision.revision_index}` : 'draft'}`;
```

Comment:
A draft version should never be available on the consumer side, and a published dataset will always have a title, so I think we can remove the condition.